components/butterfly/src/protocol/newscast.rs

```diff
@@ -32,7 +32,9 @@ impl fmt::Display for RumorType {
 
 impl From<CDeparture> for Rumor {
     fn from(value: CDeparture) -> Self {
-        let payload = Departure { member_id: Some(value.member_id), };
+        let exp = value.expiration.for_proto();
+        let payload = Departure { member_id:  Some(value.member_id),
+                                  expiration: Some(exp), };
```

Comment:
If departures are never supposed to expire, that seems like something we should enforce at the type level. Being able to set it to something here seems like we open ourselves up to confusion and inconsistency.

(repeat this for all the other non-expiring rumors)

Reply:
Can you elaborate a bit more on this?  I'm not following you on how this opens us up for confusion and inconsistency.

I'm also not entirely clear how we would enforce this at the type level.  I'll think some more on this, but if you had specific thoughts, I'd love to hear them.

Reply:
Circling back on this, I'm going to try @baumanj 's suggestion of changing `RumorExpiration(DateTime<Utc>)` to be `RumorExpiration(Option<DateTime<Utc>>)`, which I think should address your point about enforcing optionality at the type level.  If not, I'd love to hear more.